lab_3_keywords_textrank/main.py

```diff
@@ -233,7 +269,17 @@ def add_edge(self, vertex1: int, vertex2: int) -> int:
                 0 if edge was added successfully, otherwise -1
         In case of vertex1 being equal to vertex2, -1 is returned as loops are prohibited
         """
-        pass
+        if vertex1 == vertex2:
+            return -1
+        for vrtx in vertex1, vertex2:
+            if vrtx not in self._matrix[0]:
+                self._matrix[0].append(vrtx)
+                self._matrix.append([0 for _ in range(len(self._matrix[0]))])
+                for i in range(1, len(self._matrix[0])):
```

Comment:
can you explain what this cycle does and how is it different from the one in the 277th line?

Reply:
wouldn't `_matrix[len(self._matrix[0])]` have one more zeros than other rows?

Reply:
in line 277: it adds into the list another ones, which contain zeros, their length starts at 1 element - [0] and then with each step, the number of elements increases by one (to the point when it reaches the length)
the one below goes through nested lists and checks if their length equals the length of the first nested list and if it doesn't (only the last one does), it adds zeros

Reply:
well, it is supposed to look like this [[vertex1, vertex2], [0, 0], [0, 0]], i'll try to check

Reply:
my list with verteces doesn't have another 0, should it?

Reply:
we've discussed this approach with colleagues and we came to the conclusion that although this approach can be functional it is advised to only use matrix to store 0 or 1, not the actual vertices 

think of a matrix in math: it does not contain names for the rows and columns, it is purely values
please, try to come up with a way to store information about position-vertex links somehow else 

possibly, i'll write a post about it in the channel... as it is a common issue